binder/helpers.py

```diff
@@ -92,17 +92,16 @@ def delete_record(dns_server, rr_list, key_name):
         keyring = transfer_key.create_keyring()
         algorithm = transfer_key.algorithm
 
-    delete_response = []
-    for current_rr in rr_list:
-        record_list = current_rr.split(".")
-        record = record_list[0]
-        domain = ".".join(record_list[1:])
-        dns_update = dns.update.Update(domain, keyring=keyring, keyalgorithm=algorithm)
-        dns_update.delete(record)
-        output = send_dns_update(dns_update, dns_server, key_name)
-
-        delete_response.append({ "description" : "Delete Record: %s" % current_rr,
-                                 "output" : output })
+    dns_update = dns.update.Update(zone_name, keyring=keyring, keyalgorithm=algorithm)
+    records = []
+    for resource_record in rr_list:
+        dns_update.delete(resource_record)
+        records.append("%s.%s" % (zone_name, resource_record))
+
+    output = send_dns_update(dns_update, dns_server, key_name)
+    delete_response = [{"description": "Deleted Records:<br/>%s" %
+                        '<br />'.join(records),
```

Comment:
I completely agree on the output formatting. It's just been a quick solution working with the current implementation of error handling in binder in general. I believe the error handling will need a complete overhaul anyway to be more precise and flexible. I intend to address that in future with another pull request.

Regarding the loss of the record-by-record accounting I'm not sure if that's worse than before, because all records which will be deleted belong to the same zone. So we know that if the key works for one of them, it works for the others too. And as the deletion of records even returns successfully if they are not existing, we don't loose information in such a case.